packages/@react-spectrum/list/docs/ListView.mdx

```diff
@@ -0,0 +1,941 @@
+please see the [drag and drop documentation](dragAndDrop.html).
+
+The example below demonstrates how to create a draggable ListView and a droppable ListView.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+import {useDragHooks, useDropHooks} from '@react-spectrum/dnd';
+import {useListData} from '@react-stately/data';
+
+function DragIntoListExample() {
+  let sourceList = useListData({
+    initialItems: [
+      {id: '1', type: 'file', name: 'Adobe Photoshop'},
+      {id: '2', type: 'file', name: 'Adobe XD'},
+      {id: '3', type: 'file', name: 'Adobe InDesign'},
+      {id: '4', type: 'file', name: 'Adobe AfterEffects'}
+    ]
+  });
+
+  let targetList = useListData({
+    initialItems: [
+      {id: '5', type: 'file', name: 'Adobe Dreamweaver'},
+      {id: '6', type: 'file', name: 'Adobe Fresco'},
+      {id: '7', type: 'file', name: 'Adobe Connect'},
+      {key: '8', type: 'file', name: 'Adobe Lightroom'}
+    ]
+  });
+
+  // Append a generated key to the item type so they can only be dragged between these two lists
+  let dragType = React.useMemo(() => `keys-${Math.random().toString(36).slice(2)}`, []);
+  let acceptedDragTypes = `${dragType}-file`;
+
+  // Update the target and source list upon successful drop
+  let onMove = (keys, target) => {
+    if (target.type === 'root') {
+      targetList.append(...keys.map(key => sourceList.getItem(key)));
+    } else {
+      switch (target.dropPosition) {
+        case 'before':
+          targetList.insertBefore(target.key, ...keys.map(key => sourceList.getItem(key)));
+          break;
+        case 'after':
+          targetList.insertAfter(target.key, ...keys.map(key => sourceList.getItem(key)));
+          break;
+      }
+    }
+    sourceList.remove(...keys);
+  };
+
+  let dragHooks = useDragHooks({
+    getItems(keys) {
+      return [...keys].map(key => {
+
+        let item = sourceList.getItem(key);
+        // Setup the info for each item that will be propagated upon drag
+        return {
+          [`${dragType}-${item.type}`]: JSON.stringify(item)
+        };
+      });
+    }
+  });
+
+  let dropHooks = useDropHooks({
+    async onDrop(e) {
+      let keysToMove = [];
+      // Only add the dropped item to the list of keys to move
+      // if the dropped item has the expected types
+      for (let item of e.items) {
+        if (item.kind === 'text' && item.types.has(acceptedDragTypes)) {
+          let type = acceptedDragTypes;
+          let {id} = JSON.parse(await item.getText(type));
+          keysToMove.push(id);
+        }
+      }
+      onMove(keysToMove, e.target);
+    },
+    getDropOperation(target, types) {
+      let typesSet = types.types ? types.types : types;
+      let draggedTypes = [...typesSet.values()];
```

Comment:
Note that this has a typescript error since the type for `types` doesn't actually have `types.type`. There is a ticket for investigating why `types` returns a Set or a object with a types property that is a Set and either fixing that or updating the typescript type.